perllib/FixMyStreet/App/Controller/Auth.pm

```diff
@@ -187,6 +188,8 @@ sub email_sign_in : Private {
         if $c->get_param('oauth_need_email') && $c->session->{oauth}{facebook_id};
     $token_data->{twitter_id} = $c->session->{oauth}{twitter_id}
         if $c->get_param('oauth_need_email') && $c->session->{oauth}{twitter_id};
+    $token_data->{oidc_id} = $c->session->{oauth}{oidc_id}
+        if $c->get_param('oauth_need_email') && $c->session->{oauth}{oidc_id};
```

Comment:
Feels like these three could be in a little loop now, given they're all the same.